### PR TITLE
PWR068: Don't restrict to `external` procedures

### PR DESCRIPTION
**Pull request by alvrogd**
Update the contents of the `README.md` so as not to restrict the check to `external` procedures.

Just like `external` procedures, `implicit` procedures are also called through implicit interfaces. Additionally, expand the already mentioned `interface` scenario with a code example.

### CODE REVIEW

**Review by ruifm — APPROVED** [review]

File: Checks/PWR068/README.md [hunk 1]
```diff
@@ -150,11 +150,55 @@ Factorial of           5 is         120
...
-> While you can manually define explicit interfaces using the `interface`
-> construct at the call site, this approach requires duplicating the procedure's
-> definition, increasing the likelihood of errors.
-
-> [!NOTE]
+> The previous example demonstrates how calls to `external` procedures are
+> performed through implicit interfaces. A similar situation arises when
+> calling an implicitly declared procedure, as shown in the following example:
+> 
+> ```fortran
```
Comment (inaki-amatria) [inline Checks/PWR068/README.md:157]: nit:
```suggestion
> ```fortran {5}
```

**Review by inaki-amatria — APPROVED** [review]
Looks great! We could also mention Fortran's interoperability with C, where, even with explicit interfaces, there is no guarantee that the call will be correct; emphasizing the importance of ensuring that everything on the Fortran side is as checkable by the compiler as possible

File: Checks/PWR068/README.md (hunk 1 above)
Comment (alvrogd) [reply Checks/PWR068/README.md]: Done! I've used the opportunity to add relevant highlights to already existing code blocks.

**Comment by alvrogd** [comment]
> emphasizing the importance of ensuring that everything on the Fortran side is as checkable by the compiler as possible

Good idea! I've included an additional note to the entry discussing the C/C++ interoperability scenario.

**Pull request merged**